job_executor/app.py

```diff
@@ -116,6 +134,8 @@ def main():
                 except Exception as e:
                     logger.error(f'{job.job_id} failed')
                     logger.exception(e)
+                    # TODO: Catch errors in Datastore
```

Comment:
Is this TODO still accurate?

Reply:
Yes it needs to be implemented in a future PR when all the rollbacks are in order. It's explained in the PR conversation 👍🏻